x-pack/plugins/lens/public/visualizations/datatable/visualization.tsx

```diff
@@ -782,15 +799,21 @@ export const getDatatableVisualization = ({
   },
 });
 
-function getDataSourceAndSortedColumns(
+function getDatasourceAndSortedColumns(
   state: DatatableVisualizationState,
-  datasourceLayers: DatasourceLayers
+  datasourceLayers: DatasourceLayers,
+  excludeHidden = false
 ) {
+  const columnMap = new Map(state.columns.map((c) => [c.columnId, c]));
   const datasource = datasourceLayers[state.layerId];
   const originalOrder = datasource?.getTableSpec().map(({ columnId }) => columnId);
   // When we add a column it could be empty, and therefore have no order
   const sortedColumns = Array.from(
     new Set(originalOrder?.concat(state.columns.map(({ columnId }) => columnId)))
-  );
+  ).flatMap((id) => {
+    const col = columnMap.get(id);
+    if (excludeHidden && col?.hidden) return [];
+    return [id];
+  });
```

Comment:
Conditionally exclude hidden columns. This is used only for the exported csv option, the inspect always shows hidden columns.

Reply:
This is now handled in `vis. getExportDatatables`